pkg/analyzer/analyze.go

```diff
@@ -0,0 +1,665 @@
+//
+// Copyright 2022 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package guacanalyze
+
+import (
+  "context"
+  "fmt"
+
+  "os"
+
+  "github.com/Khan/genqlient/graphql"
+  "github.com/dominikbraun/graph"
+  "github.com/dominikbraun/graph/draw"
+  model "github.com/guacsec/guac/pkg/assembler/clients/generated"
+  "github.com/guacsec/guac/pkg/assembler/helpers"
+
+  "github.com/olekukonko/tablewriter"
+  "github.com/spf13/cobra"
+
+  "k8s.io/apimachinery/pkg/util/rand"
+)
+
+const PRINT_MAX = 20
+
+type HighlightedDiff struct {
+  MissingAddedRemovedLinks [][]string
+  MissingAddedRemovedNodes []string
+  MetadataMismatch []string
+}
+
+type Node struct {
+  ID string
+  Attributes map[string]interface{}
+  color string
+}
+
+func NodeHash(n *Node) string {
+  return n.ID
+}
+
+func SetNodeAttribute(g graph.Graph[string, *Node],ID, key string, value interface{}){
+  var (
+    err error
+    node *Node
+  )
+  if node, err = g.Vertex(ID); err !=  nil {
+    fmt.Println("Error setting node attribute", err)
+    os.Exit(1)
```

Comment:
Shouldn't have any os.Exit in a package under `pkg/`, assume this is a library that can be called from anywhere. Propagate errors back to the caller.